internal/exporter/pulsarexporter/pulsar_exporter.go

```diff
@@ -17,91 +17,94 @@ package pulsarexporter
 import (
 	"context"
 	"fmt"
-	"log"
-	"time"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
 
 	"github.com/apache/pulsar-client-go/pulsar"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/pdata/pmetric"
-	"go.uber.org/zap"
 )
 
 var errUnrecognizedEncoding = fmt.Errorf("unrecognized encoding")
 
-// pulsarMetricsProducer uses sarama to produce metrics messages to pulsar
-type pulsarMetricsProducer struct {
+// pulsarMetricsExporter produce metrics messages to pulsar
+type pulsarMetricsExporter struct {
+	client    pulsar.Client
 	producer  pulsar.Producer
```

Comment:
This is nice because `pulsar.Producer` is actually an interface. That means that we'll be able to swap out the implementation with our own fake one for unit tests. (The current unit tests in pulsar_exporter_test.go try to connect to a real Pulsar instance.)